ssts/segmentation/segmentation_gmm.py

```diff
@@ -0,0 +1,185 @@
+        neighborhoods around each pixel will be used. These neighborhoods will be flattened into vectors
+        and fed through the dimensionality reduction step. If `padding` is zero, no neighborhood information
+        will be used. The `embedding_dim` value will determine whether to use dimensionality reduction and,
+        if so, what the embedded dimension should be. The dimensionality step will be performed prior to
+        fitting the Gaussian mixture model.
+
+        Args:
+            n_components (int): Number of components (i.e. segments) to learn in the GMM.
+            normalize (bool): Normalize the material properties before processing.
+            embedding_dim (int): Dimension used in dimensionality reduction.
+            padding (int): Padding used in sliding window. When set to zero, no neighbor information
+                will be used. The neighbor values are flattened and fed into the PCA routine.
+            zscale (bool): Whether to use z-score scaling (True) or scale by property maxima (False).
+        """
+        self.normalize = normalize
+        self.embedding_dim = embedding_dim
+        self.n_components = n_components
+        self.padding = padding
+        self.zscale = zscale
+
+        self.gmm = None
+        self.pca = None
+        self.sst = None
+
+
+    def fit(self, data, outliers=None):
+        """
+        Performs Gaussian mixture model segmentation using
+        the provided settings.
+
+        Args:
+            data (NumPy Array): Material properties array of shape (height, width, n_properties)
+            outliers (NumPy Array): Binary array indicating outliers of shape (height, width)
+
+        Returns:
+            self (SegmenterGMM)
+        """
+        if len(data.shape) != 3:
+            logger.warning("Data arrays must be of shape (height, width, n_properties).")
+            return None
+
+        if outliers is not None and len(outliers.shape) != 2:
+            logger.warning("Outlier arrays must be of shape (height, width).")
+            return None
+
+        h, w, c = data.shape
+        n = h * w
+
+        if self.padding > 0:
+            data = SegmenterGMM.get_windows(data, padding=self.padding)
+        else:
+            data = data.reshape(n, c)
+
+        if outliers is not None:
+            outliers = outliers.flatten()
+
+        data = SegmenterGMM.remove_outliers(data, outliers)
+
+        if self.normalize:
+            if self.zscale:
+                self.sst = StandardScaler()
+                data = self.sst.fit_transform(data)
+            else:
+                data = SegmenterGMM.normalize_by_max(data)
+
+        if self.embedding_dim is not None:
+            self.pca = PCA(n_components=self.embedding_dim)
+            data = self.pca.fit_transform(data)
+
+        self.gmm = GaussianMixture(n_components=self.n_components, covariance_type="full")
+        self.gmm.fit(data)
+
+        return self
```

Comment:
Probably don't need this. Should we replace with pass?

Reply:
I believe this is what scikit does. It allows you to call
`SegmenterGMM.fit(X)` and get back the fitted segmenter